src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -72,7 +74,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         if (e.OriginalKey() == winrt::Windows::System::VirtualKey::Enter)
         {
             auto const state = CoreWindow::GetForCurrentThread().GetKeyState(winrt::Windows::System::VirtualKey::Shift);
-            if (state == CoreVirtualKeyStates::Down || state == (CoreVirtualKeyStates::Locked | CoreVirtualKeyStates::Down))
+            if (WI_IsFlagSet(state, CoreVirtualKeyStates::Down) || state == (CoreVirtualKeyStates::Locked | CoreVirtualKeyStates::Down))

```

Comment:
`WI_IsFlagSet` is sufficient for this check. You don't need to do the other one.

`WI_IsFlagSet(state, Down)` asks, "is `Down` present in `state`?" even if `state` contains multiple values. This is the reason people use bitwise enumerators -- it lets them store multiple values in the same storage location and check them individually without having to check every single possible state combination.

Reply:
changed